include/deal.II/numerics/fe_field_convolution_function.h

```diff
@@ -0,0 +1,280 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_fe_field_convolution_function_h
+#define dealii_fe_field_convolution_function_h
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/tensor.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/mapping_q1.h>
+
+#include <deal.II/grid/grid_tools_cache.h>
+
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/meshworker/scratch_data.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+  /**
+   * This is an approximated interpolation function for the given dof handler
+   * and the given solution vector, that actually performs a convolution with
+   * a specified kernel in order to compute its values.
+   *
+   * In particular, the evaluation of this function at a point $x$ is computed
+   * as:
+   *
+   * \f[
+   * F(x) = \int_{\Omega} f(y) K^{\epsilon}(x-y) dy
+   * \f]
+   * where $f(y)$ is the finite element field function constructed using the
+   * vector and the dof handler passed at construction time.
```

Comment:
* where $f(y)$ is the finite element field function that corresponds to the
￼   * vector and the dof handler passed to the constructor of this class at construction time.

But I think it would be useful to use `f_h(y)` or maybe `w_h(y)` to make it visually clear that the object corresponds to a discrete function.